kubera/dmaas/credentials/add_credential.go

```diff
@@ -0,0 +1,128 @@
+package credentials
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+
+	"github.com/mayadata-io/kubera-api-testing/kubera/common"
+
+	"github.com/go-resty/resty/v2"
+	auth "github.com/mayadata-io/kubera-api-testing/kubera/authentication"
+	"github.com/mayadata-io/kubera-api-testing/kubera/connection"
+)
+
+// CloudCredential conaitns all parameters which should be passed to add cloud credential
+type CloudCredential struct {
+	Name       string   `json:"name"`
+	ProjectID  string   `json:"projectId"`
+	ProviderID string   `json:"providerId"`
+	Cred       CredType `json:"credential"`
+}
+
+// CredType is cloud provider access id and secret key
+type CredType struct {
+	AccessID  string `json:"access_id"`
+	SecretKey string `json:"secret_key"`
+}
+
+// CredAdding represent the fields requried to add a credential
+type CredAdding struct {
+	CloudName string
+	CredName  string
+	Username  string
+	Password  string
+}
+
+// NewCredentialsAdder return fields value requried to add a credential
+func NewCredentialsAdder(cred CredAdding) *CredAdding {
+	return &CredAdding{
+		CloudName: cred.CloudName,
+		CredName:  cred.CredName,
+		Username:  cred.Username,
+		Password:  cred.Password,
+	}
+}
+
+// Adding backup provider credential
+func (a *CredAdding) addCloudCredential() (*resty.Response, *auth.APIError) {
+	conn := connection.Get()
+	fetcher := auth.NewTokenFetcher(auth.TokenFetchingConfig{
+		Connection: conn,
+	})
+	tkn, err := fetcher.Fetch()
+	if err != nil {
+		log.Fatalf("Expected no error got=%s: \nFetcher=%s: \nToken=%s", err, fetcher, tkn)
```

Comment:
how about indenting the code
```go
log.Fatalf(
  "Expected no error got=%s: \nFetcher=%s: \nToken=%s", 
  err, 
  fetcher, 
  tkn,
)
```

Reply:
Let us avoid validations & error handling in the packages.
We shall invoke t.Fatalf in _test.go files only.